pkgs/extension_discovery/lib/extension_discovery.dart

```diff
@@ -0,0 +1,244 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// Find extensions with [findExtensions].
+library;
+
+import 'dart:collection' show UnmodifiableListView;
+import 'dart:convert' show jsonDecode;
+import 'dart:io' show File, IOException;
+import 'dart:isolate' show Isolate;
+
+import 'src/io.dart';
+import 'src/package_config.dart';
+import 'src/registry.dart';
+
+export 'src/package_config.dart' show PackageConfigException;
+
+/// Information about an extension for target package.
+final class Extension {
+  /// Name of the package providing an extension.
+  final String package;
+
+  /// Absolute path to the package root.
+  ///
+  /// This folder usually contains a `lib/` folder and a `pubspec.yaml`
+  /// (assuming dependencies are fetched using the pub package manager).
+  final Uri rootUri;
```

Comment:
for both `rootUri` and `packageUri` can you give an example of what these values be so it is clear how they are different?

Reply:
Will include a link to:
https://github.com/dart-lang/language/blob/main/accepted/2.8/language-versioning/package-config-file-v2.md